nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/QueryDataCache.java

```diff
@@ -0,0 +1,32 @@
+package io.github.nucleuspowered.nucleus.api.query;
+
+/**
+ * A data cache which stores objects constructed from data retrieved from a query.
+ *
+ * @param <K> The type of the key used to retrieve cached objects.
+ * @param <V> The type of the object being cached.
+ */
+public interface QueryDataCache<K, V> {
```

Comment:
I put it in the API to allow people who wish to implement their own caching system to do so, some people may not want to use the systems provided.